### PR TITLE
Charmis/improvement changes

### PR DESCRIPTION
**Pull request by CharmisG**

### CODE REVIEW

File: react-native-in-app-payments-quickstart/app/components/CommonAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+const FailIcon = require('../images/failIcon.png');
+const SuccessIcon = require('../images/sucessIcon.png');
+
+AddressView.propTypes = {
+  title: PropTypes.string.isRequired,
+  description: PropTypes.string.isRequired,
+};
+
+export default function AddressView({ title, description, status, isVisible, onDialogClick }: { title: string, description: string, status: boolean, isVisible: boolean, onDialogClick: Function }) {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/components/CommonAlert.tsx:28]: is AddressView the right name for this component? The file name is CommonSlert.tsx

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 2]
```diff
@@ -180,24 +194,24 @@ export default function HomeScreen() {
...
-        showAlert('Error processing GooglePay payment', error.message);
+        setValue('Error processing GooglePay payment', error.message, false);
       }
     } else {
       printCurlCommand(cardDetails.nonce, SQUARE_APP_ID);
-      showAlert(
+      setValue(
         'Nonce generated but not charged',
-        'Check your console for a CURL command to charge the nonce, or replace CHARGE_SERVER_HOST with your server host.',
+        'Check your console for a CURL command to charge the nonce, or replace CHARGE_SERVER_HOST with your server host.', false
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:208]: should this be in a new line? 

We should celebrate the developer made this far. Showing a failure icon isn't encouraging. Can we make it a success message?

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 3]
```diff
@@ -209,21 +223,20 @@ export default function HomeScreen() {
...
         await chargeCardNonce(cardDetails.nonce);
         SQIPCardEntry.completeCardEntry(() => {
-          showAlert(
-            'Your order was successful',
-            'Go to your Square dashbord to see this order reflected in the sales tab.',
-          );
+          console.log(JSON.stringify(cardDetails));
+          var cardData = cardDetails.card;
+          setValue('Congratulation, Your order was successful',
+            'Go to your Square dashbord to see this order reflected in the sales tab.', true)
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:229]: should `true` be in a new line?

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 4]
```diff
@@ -608,14 +620,32 @@ export default function HomeScreen() {
     }
   };
 
+  const showValue = (title, description) => {
+    return (
+      <CommonAlert title={title} description={description} status={status} isVisible={showingDialogSheet} onDialogClick={onDialogClick} />
+    )
+  }
+
+  const setValue = (title, description, status) => {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:629]: Do we have a better name fro this method?

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 5]
```diff
@@ -608,14 +620,32 @@ export default function HomeScreen() {
     }
   };
 
+  const showValue = (title, description) => {
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:623]: we need a better name for this method.

**Review by hukid — CHANGES_REQUESTED** [review]

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 5 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:623]: Done

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 4 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:629]: Done

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 3 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:229]: Done

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx (hunk 2 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:208]: Done

File: react-native-in-app-payments-quickstart/app/components/CommonAlert.tsx (hunk 1 above)
Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/components/CommonAlert.tsx:28]: Done

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx [hunk 6]
```diff
@@ -336,7 +344,8 @@ export default function HomeScreen() {
         );
         setValue(
           'Congratulation, Your order was successful',
-          'Go to your Square dashbord to see this order reflected in the sales tab.', true
+          'Go to your Square dashbord to see this order reflected in the sales tab.',
+          true
         );
       } catch (error: any) {
         setValue('Error processing card payment', error.message, false);
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx:351]: Rename this `setValue`?

Comment (CharmisG) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx]: Done

**Pull request merged**